File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/management/DeviceCodeRecord.java
package com.enterpriseagentos.backendjava.domain.features.management;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public class DeviceCodeRecord {
    public UUID id;
    public String deviceCode;
    public String userCode;
    public UUID userId;
    public DeviceCodeStatus status;
    public String runnerName;
    public Instant expiresAt;
    public Instant createdAt;
    public Instant lastPolledAt;
    public UserRecord user;
}

File: apps/backend/src/Features/Management/Domain/DeviceCodeRecord.cs
namespace OffceOs.Domain.Features.Management;

public sealed class DeviceCodeRecord
{
    public Guid Id { get; init; } = Guid.NewGuid();

    [Required, MaxLength(64)]
    public string DeviceCode { get; init; } = string.Empty;

    [Required, MaxLength(16)]
    public string UserCode { get; init; } = string.Empty;

    public Guid? UserId { get; set; }

    public DeviceCodeStatus Status { get; set; } = DeviceCodeStatus.Pending;

    [MaxLength(200)]
    public string? RunnerName { get; set; }

    public DateTime ExpiresAt { get; init; }
    public DateTime CreatedAt { get; init; } = DateTime.UtcNow;
    public DateTime? LastPolledAt { get; set; }

    public UserRecord? User { get; set; }
}
